pallets/runtime/tests/src/utility_test.rs

```diff
@@ -228,3 +231,91 @@ fn _relay_unhappy_cases() {
         Error::InvalidNonce
     );
 }
+
+#[test]
+fn batch_secondary_with_permissions_works() {
+    ExtBuilder::default()
+        .build()
+        .execute_with(batch_secondary_with_permissions);
+}
+
+fn batch_secondary_with_permissions() {
```

Comment:
Overall this test could use some comments for the broad strokes "paragraphing" it into sections.